clients/src/main/java/org/apache/kafka/common/internals/KafkaFutureImpl.java

```diff
@@ -192,6 +192,14 @@ public T get(long timeout, TimeUnit unit) throws InterruptedException, Execution
     public T getNow(T valueIfAbsent) throws ExecutionException {
         try {
             return completableFuture.getNow(valueIfAbsent);
+        } catch (CancellationException e) {
+            // In Java 23, When a CompletableFuture is cancelled, getNow() will throw a CompletionException wrapping a 
+            // CancellationException. whereas in Java < 23, it throws a CompletionException directly.
+            if (e.getCause() instanceof CancellationException) {
+                throw (CancellationException) e.getCause();
+            } else {
+                throw new CancellationException(e.getMessage());
```

Comment:
why not throwing `e.getCause` directly?

Reply:
The method exception signature doesn't have `Throwable`, so we can't  throw `e.getCause()`.
but we can throw `CancellationException e` directory.